OsmAnd/src/net/osmand/plus/mapcontextmenu/controllers/AmenityMenuController.java

```diff
@@ -69,7 +71,8 @@ public AmenityMenuController(@NonNull MapActivity mapActivity,
 					new MapMarkerMenuController(mapActivity, marker.getPointDescription(mapActivity), marker);
 			leftTitleButtonController = markerMenuController.getLeftTitleButtonController();
 			rightTitleButtonController = markerMenuController.getRightTitleButtonController();
-		} else if (amenity.getSubType().equals(ROUTE_ARTICLE_POINT)) {
+		} else if (amenity.getSubType().equals(ROUTE_ARTICLE_POINT) ||
```

Comment:
Can subtype be null? in isRouteTrack/isRoutePoint, you added checks in other places without it. Also, it is better to use CONSTANT.equals() to avoid unnecessary npe. 

+we can introduce local subtype variable & pass it to openTrack

Reply:
13a508bf65